src/main/groovy/actions/TransferActions.groovy

```diff
@@ -4,41 +4,102 @@
+                materialType, sourceMaterial, copyMetadata)
         }
-        destinationMaterials = MaterialActions.postMaterials(destinationMaterials)
+        destinationMaterials = postNewMaterials(destinationMaterials)
 
         destinationMaterials.each { material ->
             def location = sourceMaterialUuidToLocation[material.parents[0].id]
-            def receptacle = destination.receptacles.find { it.location == location }
+            def receptacle = destinationLabware.receptacles.find { it.location == location }
             receptacle.materialUuid = material.id
         }
 
-        LabwareActions.updateLabware(destination)
+        updateLabware(destinationLabware)
+    }
+
+    def static split(Labware sourceLabware, Labware destinationLabware, MaterialType materialType,
+        List<String> destinationLocations, List<String> copyMetadata = []) {
+
+        def destinationLabwareLocations =
+            destinationLabware.receptacles.collect { it.location.name }
+        def missingLocations = new ArrayList<String>()
+        destinationLocations.each { destinationLocation ->
+            if (!destinationLabwareLocations.contains(destinationLocation)) {
+                missingLocations.add(destinationLocation)
+            }
+        }
+        if (missingLocations.size > 0) {
+            throw new TransferException("The following locations missing from the destination labware: ${missingLocations.join(', ')}")
+        }
+
+        def sourceMaterial = getMaterialsFromLabware(sourceLabware.materialUuids())
+
+        def destinationMaterials = new ArrayList<>()
+        def materialsNameByDestinationLocation = new HashMap<String, String>()
+        destinationLocations.each {
+            destinationMaterials.add(
+                createNewChildMaterial("${destinationLabware.barcode}_$it",
+                    materialType, sourceMaterial, copyMetadata)
+            )
+            materialsNameByDestinationLocation.put(it, "${destinationLabware.barcode}_$it")
+        }
+        destinationMaterials = postNewMaterials(destinationMaterials)
+
+        materialsNameByDestinationLocation.each { location, materialName ->
+            def material = destinationMaterials.find { it.name == materialName }
+            def receptacle = destinationLabware.receptacles.find { it.location.name == location }
+            receptacle.materialUuid = material.id
+        }
+
+        updateLabware(destinationLabware)
+    }
+
+    private static getMaterialsFromLabware(materialUuids) {
+        MaterialActions.getMaterials(materialUuids)
+    }
+
+    private static createNewChildMaterial(materialName, type, sourceMaterial, copyMetadata) {
+        new Material(
+            name: materialName,
+            materialType: type,
+            metadata: sourceMaterial.metadata.findAll { it.key in copyMetadata },
+            parents: [sourceMaterial]
+        )
+    }
+
+    private static postNewMaterials(destinationMaterials) {
```

Comment:
We are using them in both actions. If we have to change them, then they are in 1 place